File: script/fix-metadata.js
/**
 * fix-metadata.js  — regex-based, handles any formatting variation
 * Run: node script/fix-metadata.js
 */
const fs = require('fs');
const path = require('path');
const BASE = path.resolve(__dirname, '..');

function patchFile(relPath, patches) {
  const filePath = path.join(BASE, relPath);
  if (!fs.existsSync(filePath)) { console.error(`❌ Not found: ${relPath}`); return; }

  let content = fs.readFileSync(filePath, 'utf8');
  let changed = false;

  for (const { find, replace, desc } of patches) {
    const before = content;
    content = content.replace(find, replace);
    if (content !== before) {
      console.log(`  ✅ ${desc}`);
      changed = true;
    } else {
      console.log(`  ⚠️  Already patched or not found: ${desc}`);
    }
  }

  if (changed) fs.writeFileSync(filePath, content, 'utf8');
}

console.log('\n📄 cutoff/page.tsx');
patchFile('src/app/college/[slug]/cutoff/page.tsx', [
  {
    desc: 'title',
    find: /title:\s*`\$\{name\} Cutoff 2026`/g,
    replace: 'title: `${name} Cutoff 2026 — Branch-wise Opening & Closing Ranks, Category Cutoffs`'
  },
  {
    desc: 'description',
    find: /description:\s*`Check out the detailed cutoff marks and admission ranks for \$\{name\}, \$\{location\}\.`/g,
    replace: 'description: `Check ${name} cutoff 2026: branch-wise opening & closing ranks, category-wise (General/OBC/SC/ST) closing ranks, merit list, and admission eligibility in ${location}. Updated for 2026.`'
  }
]);

console.log('\n📄 placement/page.tsx');
patchFile('src/app/college/[slug]/placement/page.tsx', [
  {
    desc: 'title',
    find: /title:\s*`\$\{name\} Placement 2026`/g,
    replace: 'title: `${name} Placement 2026 — Average Package, Top Recruiters & Branch-wise Salary`'
  },
  {
    desc: 'description',
    find: /description:\s*`Explore the latest placement statistics, highest packages, and top recruiters for \$\{name\}, \$\{location\}\.`/g,
    replace: 'description: `${name} placement report 2026: average CTC, highest salary package, top recruiting companies, branch-wise placement stats, and internship data for the ${location} campus. Full report inside.`'
  }
]);

console.log('\n📄 admission/page.tsx');
patchFile('src/app/college/[slug]/admission/page.tsx', [
  {
    desc: 'title',
    find: /title:\s*`\$\{name\} Admission 2026`/g,
    replace: 'title: `${name} Admission 2026 — Application Form, Eligibility, Dates & Process`'
  },
  {
    desc: 'description',
    find: /description:\s*`Get complete details on the admission process, dates, and eligibility for \$\{name\}, \$\{location\}\.`/g,
    replace: 'description: `${name} admission 2026: application form dates, eligibility criteria, fees, entrance exams accepted, and step-by-step process. Apply now for 2026-27 academic year in ${location}.`'
  }
]);

console.log('\n📄 college main page.tsx');
patchFile('src/app/college/[slug]/page.tsx', [
  {
    desc: 'title',
    find: /title:\s*`\$\{name\}: Fees, Admission 2026, Courses, Cutoff, Ranking, Placement`/g,
    replace: 'title: `${name} ${location} — Fees 2026, Admission, Courses, Cutoff & Placement`'
  },
  {
    desc: 'description',
    find: /description:\s*`\$\{name\} is an institute in \$\{location\}\. Read for details on \$\{name\} Fees, Admission 2026, Courses, Cutoff, Ranking and Placement\.`/g,
    replace: 'description: `Explore ${name} in ${location}: course fees 2026, admission process, available courses, cutoff marks, rankings, placement stats, and student reviews. Compare and apply for 2026 admissions.`'
  },
  {
    desc: 'openGraph description',
    find: /description:\s*`\$\{name\} is an institute in \$\{location\}\.`/g,
    replace: 'description: `Explore ${name} in ${location}: course fees 2026, admission process, available courses, cutoff marks, rankings, placement stats, and student reviews. Compare and apply for 2026 admissions.`'
  }
]);

console.log('\n📄 blogs/[slug]/page.tsx');
patchFile('src/app/blogs/[slug]/page.tsx', [
  {
    desc: 'title (remove | EduNext suffix)',
    find: /title:\s*`\$\{blog\.title\} \| EduNext`/g,
    replace: 'title: blog.title'
  },
  {
    desc: 'description fallback',
    find: /description:\s*blog\.excerpt \|\| `Read \$\{blog\.title\}`/g,
    replace: 'description: blog.excerpt || `${blog.title} — Complete guide on EduNext.`'
  }
]);

console.log('\n✨ Done! Now run:');
console.log('git add src/app/college/\\[slug\\]/cutoff/page.tsx src/app/college/\\[slug\\]/placement/page.tsx src/app/college/\\[slug\\]/admission/page.tsx src/app/college/\\[slug\\]/page.tsx src/app/blogs/\\[slug\\]/page.tsx');
console.log('git commit -m "fix: improve metadata titles/descriptions for CTR optimization"');
console.log('git push origin main');

File: script/fix-metadata.py
#!/usr/bin/env python3
import os, sys

base = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))

patches = [
    {
        'file': 'src/app/college/[slug]/cutoff/page.tsx',
        'replacements': [
            (
                'title: `${name} Cutoff 2026`,',
                'title: `${name} Cutoff 2026 — Branch-wise Opening & Closing Ranks, Category Cutoffs`,'
            ),
            (
                'description: `Check out the detailed cutoff marks and admission ranks for ${name}, ${location}.',
                'description: `Check ${name} cutoff 2026: branch-wise opening & closing ranks, category-wise (General/OBC/SC/ST) closing ranks, merit list, and admission eligibility in ${location}. Updated for 2026.'
            ),
        ]
    },
    {
        'file': 'src/app/college/[slug]/placement/page.tsx',
        'replacements': [
            (
                'title: `${name} Placement 2026`,',
                'title: `${name} Placement 2026 — Average Package, Top Recruiters & Branch-wise Salary`,'
            ),
            (
                'description: `Explore the latest placement statistics, highest packages, and top recruiters for ${name}, ${location}.',
                'description: `${name} placement report 2026: average CTC, highest salary package, top recruiting companies, branch-wise placement stats, and internship data for the ${location} campus. Full report inside.'
            ),
        ]
    },
    {
        'file': 'src/app/college/[slug]/admission/page.tsx',
        'replacements': [
            (
                'title: `${name} Admission 2026`,',
                'title: `${name} Admission 2026 — Application Form, Eligibility, Dates & Process`,'
            ),
            (
                'description: `Get complete details on the admission process, dates, and eligibility for ${name}, ${location}.',
                'description: `${name} admission 2026: application form dates, eligibility criteria, fees, entrance exams accepted, and step-by-step process. Apply now for 2026-27 academic year in ${location}.'
            ),
        ]
    },
    {
        'file': 'src/app/college/[slug]/page.tsx',
        'replacements': [
            (
                'title: `${name}: Fees, Admission 2026, Courses, Cutoff, Ranking, Placement`,',
                'title: `${name} ${location} — Fees 2026, Admission, Courses, Cutoff & Placement`,'
            ),
            (
                'description: `${name} is an institute in ${location}. Read for details on ${name} Fees, Admission 2026, Courses, Cutoff, Ranking and Placement.',
                'description: `Explore ${name} in ${location}: course fees 2026, admission process, available courses, cutoff marks, rankings, placement stats, and student reviews. Compare and apply for 2026 admissions.'
            ),
            (
                'description: `${name} is an institute in ${location}.',
                'description: `Explore ${name} in ${location}: fees 2026, courses, cutoff, rankings, placements and student reviews.'
            ),
        ]
    },
    {
        'file': 'src/app/blogs/[slug]/page.tsx',
        'replacements': [
            (
                'title: `${blog.title} | EduNext`,',
                'title: blog.title,'
            ),
            (
                'description: blog.excerpt || `Read ${blog.title}`,',
                'description: blog.excerpt || `${blog.title} — Complete guide on EduNext.`,'
            ),
        ]
    },
]

all_good = True
for patch in patches:
    filepath = os.path.join(base, patch['file'])
    print(f'\n📄 {patch["file"]}')
    if not os.path.exists(filepath):
        print(f'  ❌ File not found')
        all_good = False
        continue
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    changed = False
    for old, new in patch['replacements']:
        if old in content:
            content = content.replace(old, new)
            print(f'  ✅ Fixed: {old[:60]}...')
            changed = True
        else:
            print(f'  ⚠️  Not found: {old[:60]}...')
    if changed:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)

print('\n✨ Done! Now commit and push:')
print('git add "src/app/college/[slug]/cutoff/page.tsx" "src/app/college/[slug]/placement/page.tsx" "src/app/college/[slug]/admission/page.tsx" "src/app/college/[slug]/page.tsx" "src/app/blogs/[slug]/page.tsx"')
print('git commit -m "fix: improve metadata titles/descriptions for CTR optimization"')
print('git push origin main')
